### DISCUSSION TITLE
React Table 7 editable Row

### DISCUSSION THREAD

**Discussion by sravan-kumar-suthari** [TanStack/table#2916] [Q&A]
hi , I am using react table 7. I want to edit the data using an edit button but the example which is provided in the GitHub or official documentation is inline editing without any button. Could you please help me out with this.?

**Comment by jorgearuv** [comment]
@sravan-kumar-suthari this could help you: https://react-table.tanstack.com/docs/examples/editable-data

**Reply by sravan-kumar-suthari** [reply]
Thank you but I want to edit the row based on checkbox . If the checkbox is checked then only it should allow to edit the data.

**Comment by jorgearuv** [comment]
@sravan-kumar-suthari sounds like just adding an extra step to this example: https://codesandbox.io/embed/github/tannerlinsley/react-table/tree/master/examples/row-selection

**Reply by sravan-kumar-suthari** [reply]
Yeah I tried but the editable example which is in official documentation is a custom hook which is written for one row n used the same hook for all rows . So I was unable to do conditional editing. The condition is either true or false for all rows

**Comment by Marco-exports** [comment]
Is there a way to allow certain columns to be edited -- while other columns stay "blocked" ?

**Reply by sravan-kumar-suthari** [reply]
Yes don't make that particular cell : editable.

**Reply by Marco-exports** [reply]
Could you give us an example ?

**Reply by ankursehdev** [reply]
Is it possible to give some examples? 
and is it possible to read the updated data afterward?

**Reply by Praveen8529** [reply]
@Marco-exports @ankursehdev, here is an example - https://codesandbox.io/s/button-click-workaround-forked-zgqe4?file=/src/App.js
![image](https://user-images.githubusercontent.com/31655178/110324043-76d49080-803b-11eb-8235-90b7eb501205.png)
